addons/web_editor/models/ir_qweb.py

```diff
@@ -49,6 +49,16 @@ def _compile_directive_snippet(self, el, options):
         div = u'<div name="%s" data-oe-type="snippet" data-oe-thumbnail="%s">' % (escape(ir_qweb.unicodifier(name)), escape(ir_qweb.unicodifier(thumbnail)))
         return [self._append(ast.Str(div))] + self._compile_node(el, options) + [self._append(ast.Str(u'</div>'))]
 
+    def _compile_directive_install(self, el, options):
+        if self.user_has_groups('base.group_system'):
+            name = el.attrib.get('t-title') or "Snippet"
+            module = self.env['ir.module.module'].search([('name', '=', el.attrib.get('t-install'))])
+            thumbnail = el.attrib.pop('t-thumbnail', "oe-thumbnail")
+            div = u'<div name="%s" data-oe-type="snippet" data-module_id="%s" class="o_snippet_install" data-oe-thumbnail="%s"><section></section></div>' % (escape(ir_qweb.unicodifier(name)), module and module.id, escape(ir_qweb.unicodifier(thumbnail)))
```

Comment:
hello @qsm-odoo ,
If there's no content in the snippet, thumbnail image won't appear for that snippet. So I've added empty section.